### PR TITLE
[SPARK-41806][SQL] Use AppendData.byName for SQL INSERT INTO by name for DSV2

### PR DESCRIPTION
**Pull request by allisonport-db**
### What changes were proposed in this pull request?

Use DSv2 AppendData.byName for INSERT INTO by name instead of reordering and converting to AppendData.byOrdinal

### Why are the changes needed?

Currently for INSERT INTO by name we reorder the value list and convert it to INSERT INTO by ordinal. Since DSv2 logical nodes have the `isByName` flag we don't need to do this. The current approach is limiting in that

- Users must provide the full list of table columns (this limits the functionality for features like generated columns see [SPARK-41290](https://issues.apache.org/jira/browse/SPARK-41290))
- It allows ambiguous queries such as `INSERT OVERWRITE t PARTITION (c='1') (c) VALUES ('2')` where the user provides both the static partition column 'c' and the column 'c' in the column list. We should check that the static partition column is not in the column list. See the added test for more detailed example.

### Does this PR introduce _any_ user-facing change?

For versions 3.3 and below:
```sql
CREATE TABLE t(i STRING, c string) USING PARQUET PARTITIONED BY (c);
INSERT OVERWRITE t PARTITION (c='1') (c) VALUES ('2')
SELECT * FROM t
```
```
+---+---+
|  i|  c|
+---+---+
|  2|  1|
+---+---+
```
For versions 3.4 and above:
```sql
CREATE TABLE t(i STRING, c string) USING PARQUET PARTITIONED BY (c);
INSERT OVERWRITE t PARTITION (c='1') (c) VALUES ('2')
```
```
AnalysisException: [STATIC_PARTITION_COLUMN_IN_COLUMN_LIST] Static partition column c is also specified in the column list.
```

### How was this patch tested?

Unit tests are added.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1115,6 +1115,11 @@
       "Star (*) is not allowed in a select list when GROUP BY an ordinal position is used."
     ]
   },
+  "STATIC_PARTITION_COLUMN_IN_COLUMN_LIST" : {
```
Comment (cloud-fan) [inline core/src/main/resources/error/error-classes.json:1118]:
```suggestion
  "STATIC_PARTITION_COLUMN_IN_INSERT_COLUMN_LIST" : {
```

**Review by cloud-fan — APPROVED** [review]
LGTM if tests pass

**Comment by allisonport-db** [comment]
@cloud-fan Updated and tests are passing. Can you merge?

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**